distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso

```diff
@@ -2506,6 +2507,68 @@ type Table
     ## Creates a text representation of the table using the CSV format.
     to_csv : Text
     to_csv self = Text.from self (Delimited_Format.Delimited delimiter=",")
+    
+    ## GROUP Standard.Base.Conversions
+       Returns a string containing an XML representation of the table.
+       Arguments:
+       - element_columns: Columns to be used as elements in the XML.
+       - attribute_columns: Columns to be used as attributes in the XML.
+       - value_column: Column to be used as the value for the row tag in the XML.
+       - root_name: The name of the root tag in the XML.
+       - row_name: The name of the row tag in the XML.
+       - on_problems: Specifies how to handle warnings if they occur, reporting
+         them as warnings by default.     
+
+       ! Error Conditions
+
+         - If a column in `element_columns`, `attribute_columns` or `value_column` is not in 
+           the input table, a `Missing_Input_Columns` is raised as an error.
+         - If any incomming columns aree not specified in one of `element_columns`, 
+           `attribute_columns` or `value_column`, a `Unused_Columns`
+           is reported according to the `on_problems` setting.
+
+       ? Example to_xml Operation
+
+         Input Table `table`:
+
+            Title                  | Author              | Price | Year 
+           ------------------------+---------------------+-------+------
+            A Tale Of Two Cities   | Charles Dickens     | 9.99  | 1859 
+            The Great Gatsby       | F. Scott Fitzgerald | 5.99  | 1925 
+
+         Result `r = t.to_xml ["Year"] ["Author", "Price"] "Title" "Books" "Book"`:
+
+            <Books>
+              <Book Author="Charles Dickens" Price="9.99">
+                A Tale Of Two Cities
+                <Year>1859</Year>
+              </Book>
+              <Book Author="F. Scott Fitzgerald" Price="5.99">
+                The Great Gatsby
+                <Year>1925</Year>
+              </Book>
+            </Books>
+    @element_columns Widget_Helpers.make_column_name_vector_selector
+    @attribute_columns Widget_Helpers.make_column_name_vector_selector
+    @value_column Widget_Helpers.make_column_name_selector
+    to_xml self (element_columns : (Vector (Integer | Text | Regex) | Text | Integer | Regex) = self.column_names) (attribute_columns : (Vector (Integer | Text | Regex) | Text | Integer | Regex) = []) (value_column : Text | Integer | Regex = "") (root_name : Text = "") (row_name : Text = "") (on_problems : Problem_Behavior = Report_Warning) -> Text =
```

Comment:
The `root_name` and `row_name` should have valid defaults (e.g. `"Table"` and `"Row"`).
Value column shouldn't allow Regex I don't think as needs to be a single match or nothing. It should probably take `Nothing` as an option meaning no column instead of `""`.